tests/MongoDB.Driver.Tests/MongoClientTests.cs

```diff
@@ -206,6 +210,33 @@ public void UsesSameClusterWhenReadPreferenceTagsAreTheSame()
             databaseNames.Should().Equal(operationResult["databases"].AsBsonArray.Select(record => record["name"].AsString));
         }
 
+        // will probably need to move this test?
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void ReadPreference_should_not_be_sent_to_standalone_server(
+            [Values(false, true)] bool async)
+        {
+            RequireServer.Check().ClusterType(ClusterType.Standalone);
+            var eventCapturer = new EventCapturer().Capture<CommandStartedEvent>(e => e.CommandName.Equals("find"));
+            using (var subject = new DisposableMongoClient(CreateDisposableClient(eventCapturer).WithReadPreference(ReadPreference.Secondary)))
```

Comment:
Double creation of `DisposableMongoClient`?

Reply:
Also, does it make sense to set `ReadPreference.Secondary` for `standalone` cluster? cc: @vincentkam

Reply:
`WithReadPreference(...)` returns an IMongoClient instead of an IDisposable, so I wrapped it to ensure that it would be the right type. I'd be happy to change it if there's a better way of assigning read preference to a disposable client, but I didn't know of one.

As for setting the read preference to Secondary, for a standalone server there is no read preference that really makes sense. @vincentkam and I discussed this offline and the intention was to check if sending a read preference of Secondary would cause any unforeseen issues, and when it didn't, we just left it. I'm not opposed to changing it, but I'm not sure the actual read preference type makes a difference for the test.

Reply:
Hm, Dima you're right, I missed that the first time I was talking this through with Katie: creating two disposable clients does incur a bit of a cost. The issue is that `WithReadPreference` returns `wrapped.WithReadPreference`. Thinking about it a bit more, we could avoid the double creation and disposal cost by adding a `ReadPreference` parameter to `CreateDisposableClient`. What do you think @ksadoff? 

As for setting `ReadPreference.Secondary`, the actual read preference doesn't make any difference I think in the sense that regardless of what the read preference is for a standalone, we shouldn't send it.